x-pack/elastic-agent/pkg/agent/cmd/container.go

```diff
@@ -112,15 +117,101 @@ all the above actions will be skipped, because the Elastic Agent has already bee
 occurs on every start of the container set FLEET_FORCE to 1.
 `,
 		Run: func(c *cobra.Command, args []string) {
-			if err := containerCmd(streams, c, flags, args); err != nil {
-				fmt.Fprintf(streams.Err, "Error: %v\n", err)
+			iscloud, err := c.Flags().GetBool("cloud")
+			if err != nil {
+				logError(streams, err)
+			}
+			if iscloud {
+				err = containerCloudCmd(streams, c, flags, args)
+			} else {
+				err = containerCmd(streams, c, flags, defaultAccessConfig())
+			}
+			if err != nil {
+				logError(streams, err)
 				os.Exit(1)
 			}
 		},
 	}
+	cmd.Flags().Bool("cloud", false, "Start container in cloud mode")
```

Comment:
Why use a command-line flag? Why not use an ENV variable? That way the entrypoint doesn't need to be adjusted.

Reply:
Nit: Maybe we should call it `elastic-cloud` to make it clear, this is specific to "our" cloud? Independent of a command line flag or ENV

Reply:
Added it as flag originally as the Elastic cloud builds its own entrypoint anyways; changed to `ELASTIC_AGENT_CLOUD` ENV now - it is easier for people who want to test it.